homeassistant/components/wilight/config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Config flow to configure WiLight."""
+import logging
+from urllib.parse import urlparse
+
+import pywilight
+
+from homeassistant.components import ssdp
+from homeassistant.config_entries import CONN_CLASS_LOCAL_PUSH, ConfigFlow
+from homeassistant.const import CONF_HOST
+
+from . import PLATFORMS
+from .const import DOMAIN  # pylint: disable=unused-import
+
+CONF_SERIAL_NUMBER = "serial_number"
+CONF_MODEL_NAME = "model_name"
+
+WILIGHT_MANUFACTURER = "All Automacao Ltda"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WiLightFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a WiLight config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the WiLight flow."""
+        self._host = None
+        self._serial_number = None
+        self._title = None
+        self._model_name = None
+        self._components = []
+        self._components_text = ""
+
+    def _self_update(self, host, serial_number, model_name):
+        self._host = host
+        self._serial_number = serial_number
+        self._title = f"WL{serial_number}"
+        self._model_name = model_name
+        self._components = pywilight.get_components_from_model(model_name)
+        self._components_text = ", ".join(self._components)
+        return self._components_text != ""
+
+    def _get_entry(self):
+        data = {
+            CONF_HOST: self._host,
+            CONF_SERIAL_NUMBER: self._serial_number,
+            CONF_MODEL_NAME: self._model_name,
+        }
+        return self.async_create_entry(title=self._title, data=data)
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered WiLight."""
+        # Filter out basic information
+        if (
+            ssdp.ATTR_SSDP_LOCATION not in discovery_info
+            or ssdp.ATTR_UPNP_MANUFACTURER not in discovery_info
+            or ssdp.ATTR_UPNP_SERIAL not in discovery_info
+            or ssdp.ATTR_UPNP_MODEL_NAME not in discovery_info
+            or ssdp.ATTR_UPNP_MODEL_NUMBER not in discovery_info
+        ):
+            return self.async_abort(reason="not_wilight")
+        # Filter out non-WiLight devices
+        if discovery_info[ssdp.ATTR_UPNP_MANUFACTURER] != WILIGHT_MANUFACTURER:
+            return self.async_abort(reason="not_wilight")
+
+        host = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname
+        serial_number = discovery_info[ssdp.ATTR_UPNP_SERIAL]
+        model_name = discovery_info[ssdp.ATTR_UPNP_MODEL_NAME]
+
+        if not self._self_update(host, serial_number, model_name):
+            return self.async_abort(reason="not_wilight")
+
+        allowed_components = PLATFORMS
```

Comment:
Please create a separate constant for `allowed_components`. We want to keep the home assistant API separate from the Wilight API as either of those may change.

Reply:
PLATFORMS is a constant that comes from __INIT__.py and not from the Wilight  API.
It serves to indicate which components are allowed by the current integration.
I used allowed_components for your suggestion.
I used the name PLATFORMS, because it was already in a template created by Homeassistant.
Would you like me to rename PLATFORMS to ALLOWED_COMPONENTS in __INIT__.py?

Reply:
This is where I see if the components available in the WiLight API are allowed by the current integration.

Reply:
The `PLATFORMS` constant contains strings for the home assistant api. `allowed_components` should contain strings for the wilight api. They must be created in different separate variables. We have to create a new constant, named eg `ALLOWED_WILIGHT_COMPONENTS`. Don't change `PLATFORMS`.

Reply:
Honestly I don't see it like that, I see a check of the components of the WiLight API (self._components) against the components of the Homeassistant API (PLATFORMS), to see if the components of the WiLight API are allowed in this Homeassistant integration. self._components is updated by the _self_update function.
I am renaming self._components to self._wilight_components and _self_update to _wilight_update for clarity.
If you really want me to create a new constant, I will, but it doesn't make sense because I will have to update PLATFORMS and the new constant equally in an upcoming PR to include new components.

Reply:
This conversation made me review the concept of this comparison and I saw that where any was, in reality it should be all.
In this way, only if ALL WiLight components are allowed for the HA integration version.
This occurs in the case of WiLight for ceiling fan, which has the light component and the fan on the same device.

         allowed_components = PLATFORMS
         component_ok = all (
             wilight_component in allowed_components
             for wilight_component in self._wilight_components
         )